devtools/cli/src/main/java/io/quarkus/cli/core/QuarkusCliVersion.java

```diff
@@ -17,18 +13,13 @@ public static String version() {
         if (version != null) {
             return version;
         }
-        final URL quarkusPropsUrl = Thread.currentThread().getContextClassLoader().getResource("quarkus.properties");
-        if (quarkusPropsUrl == null) {
-            throw new RuntimeException("Failed to locate quarkus.properties on the classpath");
-        }
+
         final Properties props = new Properties();
-        ClassPathUtils.consumeAsPath(quarkusPropsUrl, p -> {
-            try (BufferedReader reader = Files.newBufferedReader(p)) {
-                props.load(reader);
-            } catch (IOException e) {
-                throw new RuntimeException("Failed to load quarkus.properties", e);
-            }
-        });
+        try {
+            props.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("quarkus.properties"));
```

Comment:
This may leave the jar locked on windows after the process terminates.

Reply:
what you could consider is getting the URL like it was originally and check for the protocol. Unless it's `file` or `jar`, you can use the URL to open the stream.

Reply:
@aloubyansky thanks for being vigilant. I think I fixed it, could you check?